File: solutions/python/sublist/1/sublist.py
"""
This exercise stub and the test suite contain several enumerated constants.

Enumerated constants can be done with a NAME assigned to an arbitrary,
but unique value. An integer is traditionally used because it’s memory
efficient.
It is a common practice to export both constants and functions that work with
those constants (ex. the constants in the os, subprocess and re modules).

You can learn more here: https://en.wikipedia.org/wiki/Enumerated_type
"""

# Possible sublist categories.
# Change the values as you see fit.
SUBLIST = -1
SUPERLIST = 1
EQUAL = 0
UNEQUAL = 100


def sublist(list_one, list_two):
    if list_one==[] and list_two==[]:
        return EQUAL

    if list_one==[] and list_two!=[]:
        return SUBLIST

    if list_two==[] and list_one!=[]:
        return SUPERLIST

    if list_one==list_two:
        return EQUAL

    str1='.'.join(str(item) for item in list_one)+'.'
    str2='.'.join(str(item) for item in list_two)+'.'

    if str1 in str2:
        return SUBLIST
    if str2 in str1:
        return SUPERLIST
    return UNEQUAL

File: solutions/dart/sublist/1/lib/sublist.dart
// Implement the Classification enum.

enum Classification { equal, sublist, superlist, unequal }

class Sublist {
  Classification sublist(List<int> listOne, List<int> listTwo) {
    if (listOne.isEmpty && listTwo.isEmpty) {
      return Classification.equal;
    }
    if (listOne.isEmpty) {
      return Classification.sublist;
    }
    if (listTwo.isEmpty) {
      return Classification.superlist;
    }

    if (listOne.toString() == listTwo.toString()) {
      return Classification.equal;
    }
    // Check for sublist relationships first
    if (containsSublist(listOne, listTwo)) {
      return Classification.superlist;
    }
    if (containsSublist(listTwo, listOne)) {
      return Classification.sublist;
    }

    // Finally check for unequal if no sublist relationship is found
    return Classification.unequal;
  }

  bool containsSublist(List<int> larger, List<int> smaller) {
    for (int i = 0; i <= larger.length - smaller.length; i++) {
      if (larger.sublist(i, i + smaller.length).toString() ==
          smaller.toString()) {
        return true;
      }
    }
    return false;
  }
}
